snapcraft/pluginhandler.py

```diff
@@ -223,11 +223,13 @@ def get_state(self, step):
     def _step_state_file(self, step):
         return os.path.join(self.statedir, step)
 
-    def _setup_stage_packages(self):
-        ubuntu = repo.Ubuntu(
+    def _fetch_stage_packages(self):
+        self.ubuntu = repo.Ubuntu(
```

Comment:
This should be done in `__init__` with a conditional

```python
if self.stage_packages:
    self.ubuntu = repo.Ubuntu(...)
```

or maybe regardless as some plugins might do crazy stuff with `stage_packages` after `__init__` :wink: